### PR TITLE
[#4981] Improvement(test): Add integration tests for Hive catalog with S3 location.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add integration tests to test hive catalog with S3 location. 

### Why are the changes needed?

To make code more robust.

Fix: #4981 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

The changes itself is tested.

### CODE REVIEW

**Comment by yuqi1129** [comment]
This PR is not ready for review until #4982 is merged.

File: dev/docker/hive/hive-site.xml [hunk 1]
```diff
@@ -42,4 +42,25 @@
...
+  <property>
+    <name>fs.s3a.endpoint</name>
+    <value>S3_ENDPOINT_ID</value>
+  </property>
+
+  <property>
+    <name>fs.s3a.aws.credentials.provider</name>
+    <value>org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider,com.amazonaws.auth.EnvironmentVariableCredentialsProvider,org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider</value>
+  </property>
+
```
Comment (jerryshao) [inline dev/docker/hive/hive-site.xml:65]: I remembered that we already merged this, why do you have this again in this PR?

Comment (yuqi1129) [reply dev/docker/hive/hive-site.xml:65]: I should not be shown in the code changes as https://github.com/apache/gravitino/pull/4980 is merged, let me check the reason.

Comment (yuqi1129) [reply dev/docker/hive/hive-site.xml]: @jerryshao Rebase the main branch solves the problem.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java [hunk 2]
```diff
@@ -71,6 +71,16 @@ public class ContainerSuite implements Closeable {
...
 
+  private static volatile GravitinoLocalStackContainer gravitinoLocalStackContainer;
+
+  /**
+   * We can share the same Hive container as Hive container with S3 contains the following
+   * differences: 1. Configuration of S3 and corresponding environment variables 2. The Hive
+   * container with S3 is Hive3 container and the Hive container is Hive2 container. There is
+   * something wrong with the hive2 container to access the S3.
+   */
+  private static volatile HiveContainer hiveContainerWithS3;
```
Comment (yuqi1129) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/ContainerSuite.java:82]: We cannot share the container even if we uniform the hive version to hive3 for the following reasons.
1. We need to ensure the s3 environment is ready before starting the Hive container so we can share it.
2. for the reason in #(1, we need to start localstack(S3 simulator) first, however we can't make sure `CatalogHiveS3IT` was the first one to execute.

**Comment by jerryshao** [comment]
I'm OK with this PR itself. But I think compared to IT, it is more valuable to add some docs to claim that this can be run on S3 and walk users to some references about how to configure Hive on S3, that should be enough.

**Comment by yuqi1129** [comment]
> I'm OK with this PR itself. But I think compared to IT, it is more valuable to add some docs to claim that this can be run on S3 and walk users to some references about how to configure Hive on S3, that should be enough.

Got it, it's not a big deal to add a document reference about it and I'll do it today.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**